lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -174,10 +177,10 @@ public TOutput get(final long timeout, final TimeUnit unit)
     }
   }
 
-  /**
-   * Called by VortexMaster to let the user know that the task completed.
-   */
-  public void completed(final TOutput result) {
+  @Private
+  @Override
+  public void completed(final int taskletId, final TOutput result) {
+    assert taskletId == this.futureTaskletId;
```

Comment:
@yunseong essentially, this enables a mapping of multiple taskletIds (which corresponds VortexFunctions that the user has passed in) that have been aggregated to a single, aggregated output. Hopefully this clears things up. Let me know if there might be a better way to design this, or if something is still unclear. Thanks!